.github/workflows/dockerpublish.yml

```diff
@@ -22,87 +22,62 @@ env:
 jobs:
   push:
     runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        registry:
+          - docker
+          - github
+
     steps:
       - uses: actions/checkout@v2
 
       - name: Build image
-        run: docker build --pull -f ./.docker/Dockerfile -t "${IMAGE_NAME}" ./.docker
+        run: docker build --pull -f ./.docker/Dockerfile -t "${{ env.IMAGE_NAME }}" ./.docker
 
-      - name: Log into docker hub registry
+      - name: Log into Docker Hub
+        if: matrix.registry == 'docker'
         run: echo "${{ secrets.DOCKER_HUB_PASSWORD }}" | docker login -u "${{ github.actor }}" --password-stdin
```

Comment:
Access to secrets will fail for forks (unless `pull_request_target` is used instead of `pull_request` as trigger).

Reply:
Totally makes sense. 👍